src/partition/repart.c

```diff
@@ -2080,16 +2080,14 @@ static void context_bar_char_process_partition(
         from = p->offset;
         to = from + p->new_size;
 
-        assert(context->end >= context->start);
-        total = context->end - context->start;
+        assert(context->total > 0);
+        total = context->total;
 
-        assert(from >= context->start);
-        assert(from <= context->end);
-        x = (from - context->start) * n / total;
+        assert(from <= total);
+        x = (from - 0) * n / total;
```

Comment:
- 0?

Reply:
The very beginning of a disk (?)

Reply:
Note that this is purely about the bar printing, so we shouldn't need to check whether partition start / end are actually "valid" (e.g. before the partition table end) here, regardless of whether the partition is foreign.